libsolidity/analysis/ContractLevelChecker.h

```diff
@@ -34,13 +37,45 @@ namespace dev
 namespace solidity
 {
 
+template <class T>
+bool hasEqualNameAndParameters(T const& _a, T const& _b)
+{
+	return _a.name() == _b.name() &&
+		FunctionType(_a).asCallableFunction(false)->hasEqualParameterTypes(
+			*FunctionType(_b).asCallableFunction(false)
+		);
+}
+
+struct LessFunction
+{
+	bool operator()(FunctionDefinition const* _a, FunctionDefinition const* _b) const
+	{
+		if (hasEqualNameAndParameters(*_a, *_b))
```

Comment:
Is this actually really needed? Isn't that covered by the cases below?

Reply:
I just looked at the `parametersAreEqual` function in more detail. It does an `operator==()` comparison with all the types and the implementations of those are actually quite diverse depending on the type. I wouldn't trust a simpel type->toString() comparison to always yield the same result.

Reply:
True, but then you should rather use the same kind of comparison instead of the ``toString`` below... otherwise you get up a messed up linear order in the end.

Reply:
Even much simpler, if you use ``richIdentifier``, which is explicitly defined to compare equal if and only if the types compare equal.

Reply:
`richIdentifier` sounds perfect for this.